### ISSUE TITLE
sail-rv32 not building

### ISSUE THREAD

**Issue by marnovandermaas** [CTSRD-CHERI/TestRIG#11]
There is a problem in the default make, which says there is a syntax error in sail-riscv. The exact error message is as follows:

```
ARCH=RV32 make -C riscv-implementations/sail-riscv c_emulator/riscv_rvfi_RV32
make[1]: Entering directory '/local/scratch/mv380/testing/TestRIG/riscv-implementations/sail-riscv'
mkdir -p generated_definitions/c
sail -dno_cast -O -Oconstant_fold -memo_z3 -c -c_include riscv_prelude.h -c_include riscv_platform.h -c_no_main model/prelude.sail model/prelude_mapping.sail model/riscv_xlen32.sail model/riscv_flen_F.sail model/prelude_mem_metadata.sail model/prelude_mem.sail model/rvfi_dii.sail model/riscv_types_common.sail model/riscv_types_ext.sail model/riscv_types.sail model/riscv_vmem_types.sail model/riscv_reg_type.sail model/riscv_freg_type.sail model/riscv_regs.sail model/riscv_pc_access.sail model/riscv_sys_regs.sail model/riscv_pmp_regs.sail model/riscv_pmp_control.sail model/riscv_ext_regs.sail model/riscv_addr_checks_common.sail model/riscv_addr_checks.sail model/riscv_misa_ext.sail model/riscv_csr_map.sail model/riscv_next_regs.sail model/riscv_sys_exceptions.sail model/riscv_sync_exception.sail model/riscv_next_control.sail model/riscv_fdext_regs.sail model/riscv_fdext_control.sail model/riscv_csr_ext.sail model/riscv_sys_control.sail model/riscv_platform.sail model/riscv_mem.sail model/riscv_pte.sail model/riscv_ptw.sail model/riscv_vmem_common.sail model/riscv_vmem_tlb.sail model/riscv_vmem_sv32.sail model/riscv_vmem_rv32.sail model/riscv_insts_begin.sail model/riscv_insts_base.sail model/riscv_insts_aext.sail model/riscv_insts_cext.sail model/riscv_insts_mext.sail model/riscv_insts_zicsr.sail model/riscv_insts_next.sail model/riscv_softfloat_interface.sail model/riscv_insts_fext.sail model/riscv_jalr_seq.sail model/riscv_insts_end.sail model/riscv_step_common.sail model/riscv_step_rvfi.sail model/riscv_decode_ext.sail model/riscv_fetch_rvfi.sail model/riscv_step.sail model/main.sail -o generated_definitions/c/riscv_rvfi_model_RV32
Syntax error:
[model/riscv_insts_begin.sail]:15:30-30
15 |val encdec : ast <-> bits(32) effect {rreg}
   |                              ^
   | current token: effect
Makefile:236: recipe for target 'generated_definitions/c/riscv_rvfi_model_RV32.c' failed
make[1]: *** [generated_definitions/c/riscv_rvfi_model_RV32.c] Error 1
make[1]: Leaving directory '/local/scratch/mv380/testing/TestRIG/riscv-implementations/sail-riscv'
Makefile:144: recipe for target 'sail-rv32' failed
make: *** [sail-rv32] Error 2

```

**Comment by arichardson** [comment]
It worked for me yesterday? Is your sail binary new enough?

**Comment by rmn30** [comment]
Most likely you need to upgrade to the latest version of sail.

**Comment by marnovandermaas** [comment]
I followed the instructions on the sail wiki and my current sail version is:
```
$ sail -v
Sail 0.10 (sail2 @ opam)
```
I also tried running the following with no result:
```
$ opam update
$ opam upgrade
$ opam reinstall sail
```
Do I need to do something else?

**Comment by rmn30** [comment]
The latest released version of sail is 0.12. Perhaps you have it pinned? What does `opam pin` say?

**Comment by marnovandermaas** [comment]
Sorry the problem was that `opam update` was not working properly. I now have sail version 0.12 and that seems to solve this issue.
